### PR TITLE
Enh/purge functions

### PR DESCRIPTION
**Pull request by ndcampbell**
Just made command into a couple of functions that is easier to follow and test.

Not sure how to make good unittests for these though, it is all just mocked away.

### CODE REVIEW

File: src/lavatory/commands/purge.py [hunk 1]
```diff
@@ -15,46 +16,69 @@
...
-    
+        all_repos = before_purge_data.keys()
+
+    apply_purge_policies(all_repos, policies_path=policies_path, dryrun=dryrun, default=default)
+    generate_purge_report(all_repos, before_purge_data)
+
+    LOG.info("Success.")
+
+
+def apply_purge_policies(all_repos, policies_path=None, dryrun=True, default=True):
```
Comment (sijis) [inline src/lavatory/commands/purge.py]: I think this should return something, even if its just `True`.

If something is captured, we can definitely use that to notify/update something about what was purged.

File: src/lavatory/commands/purge.py [hunk 2]
```diff
@@ -15,46 +16,69 @@
...
-    plugin_source = setup_pluginbase(extra_policies_path=policies_path)
-
-    before_repo_data = artifactory.list()
+    before_purge_data = artifactory.list()
     if repo:
         all_repos = repo
     else:
-        all_repos = before_repo_data.keys()
-    
+        all_repos = before_purge_data.keys()
```
Comment (sijis) [inline src/lavatory/commands/purge.py]: I'd change `all_repos` to `selected_repos`. The name makes me believe I'm getting all the repos, where, in reality I may not.

File: src/lavatory/commands/purge.py [hunk 3]
```diff
@@ -15,46 +16,69 @@
...
+
+
+def generate_purge_report(purged_repos, before_purge_data):
+    """Generates a performance report based on deleted artifacts.
+
+    Args:
+        purged_repos (list): List of repos that had policy applied.
+        before_purge_data (dict): Data on the state of Artifactory before purged artifacts
+    """
+    LOG.info("\nPurging Performance:")
```
Comment (sijis) [inline src/lavatory/commands/purge.py]: I'm not sure you need the `\n`.

**Comment by ndcampbell** [comment]
@sijis added your feedback. The unittests are hard with these functions, but they did already catch a couple errors, like incompatibility with Python 3.4. Not sure how I can make them better though.

**Pull request merged**